### PR TITLE
Add pagination and swagger

### PR DESCRIPTION
**Pull request by teract10s**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,30 @@
...
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
```
Comment (NazarSmith228) [inline pom.xml]: duplicates

File: pom.xml [hunk 2]
```diff
@@ -58,6 +58,30 @@
...
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (NazarSmith228) [inline pom.xml]: version is redundant here

File: pom.xml [hunk 3]
```diff
@@ -58,6 +58,30 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-maven-plugin</artifactId>
```
Comment (NazarSmith228) [inline pom.xml]: this is a plugin - pls remove it from dependencies

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -1,38 +1,75 @@
...
 public class BookController {
-    @Autowired
-    private BookService bookService;
+    private final BookService bookService;
 
     @GetMapping("/")
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/controller/BookController.java]: you may also find `@PageableDefault` annotation useful

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,11 +1,22 @@
...
+import lombok.Data;
 
-public record CreateBookRequestDto(String title,
-                                   String author,
-                                   String isbn,
-                                   BigDecimal price,
-                                   String description,
-                                   String coverImage) {
+@Data
+public class CreateBookRequestDto {
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java]: what is the point from replacing record with a class?

File: src/main/java/com/example/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFound(EntityNotFoundException ex) {
+        return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
```

File: src/main/java/com/example/bookstore/mapper/BookMapper.java [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl")
 public interface BookMapper {
     BookDto toDto(Book book);
 
     Book toBook(CreateBookRequestDto requestDto);
+
+    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
+    void upgradeBook(CreateBookRequestDto requestDto, @MappingTarget Book book);
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/mapper/BookMapper.java]: terminology matters :)
```suggestion
    void updateBook(CreateBookRequestDto requestDto, @MappingTarget Book book);
```

File: src/main/resources/liquibase.properties [hunk 8]
```diff
@@ -0,0 +1,5 @@
+url=jdbc:h2:mem:book_store
```
Comment (NazarSmith228) [inline src/main/resources/liquibase.properties]: why Liquibase works with H2, while Spring Boot works with MySQL?

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

File: src/main/resources/liquibase.properties (hunk 8 above)
Comment (teract10s) [reply src/main/resources/liquibase.properties]: I had a problem when I use MySQL here, the project can't be built in GitHub. But locally everything is okay, and I use MySQL. So i have to use H2 here to complete CI test

File: pom.xml (hunk 1 above)
Comment (liliia-ponomarenko) [reply pom.xml]: not fixed

File: pom.xml (hunk 2 above)
Comment (liliia-ponomarenko) [reply pom.xml]: not fixed

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Not all previous comments were fixed

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**